snuba/datasets/storages/transactions.py

```diff
@@ -198,16 +169,11 @@ def transactions_migrations(
     migration_function=transactions_migrations,
 )
 
+
 storage = WritableTableStorage(
     storage_key=StorageKey.TRANSACTIONS,
     storage_set_key=StorageSetKey.TRANSACTIONS,
     schemas=StorageSchemas(read_schema=schema, write_schema=schema),
-    table_writer=TransactionsTableWriter(
```

Comment:
Yes. I think that instantiating the table writer within WritableTableStorage is better because it enforces the 1:1 relationship between a writable storage and a table writer instance. The way it was before allowed you to pass the same writer to 2 storages, but that's not actually valid and things would get weird even if you don't really get an error.

I think there are some related benefits of this. For example, the new structure prevents you from passing a different write schema to the table writer and the storage, which was possible before. It means that the storage set key used to determine the cluster of the writer can be derived from the storage set key defined in the storage.